File: Sort-3-Hindex/Hindex2.cpp
#include <bits/stdc++.h>

using namespace std;

void print(vector<int> citations)
{
    for (int i = 0; i < citations.size(); i++)
        cout << citations[i] << ' ';
    cout << endl;
}

bool compare(const int &a, const int &b)
{
    return a < b;
}

int solution(vector<int> citations)
{
    sort(citations.begin(), citations.end(), compare);
    int n = citations.size();
    for (int i = 0; i < n; i++)
    {
        if (citations[i] >= n - i)
            return n - i;
    }
    return 0;
}

int main()
{
    vector<int> citations = {3, 0, 6, 1, 5};
    int result;
    result = solution(citations);
    cout << result << endl;

    return 0;
}

File: Sort-3-Hindex/Hindex2.py
def solution(citations):
    citations.sort(reverse=True)
    answer = max(map(min, enumerate(citations, start=1)))
    return answer


citations = [9, 8, 8, 8, 8]
# citations = [9, 8, 8, 8, 8, 1, 1, 1, 1, 1]
# citations = [3, 0, 6, 1, 5]
# citations = [3, 3, 3, 3, 3]

# return = 3
result = solution(citations)
print(result)
